util/topsql/reporter/reporter.go

```diff
@@ -209,14 +229,90 @@ func (tsr *RemoteTopSQLReporter) Close() {
 	tsr.client.Close()
 }
 
+func addEvictedCPUTime(collectTarget map[string]*dataPoints, timestamp uint64, totalCPUTimeMs uint32) {
+	if totalCPUTimeMs == 0 {
+		return
+	}
+	others, ok := collectTarget[keyOthers]
+	if !ok {
+		others = &dataPoints{}
+		collectTarget[keyOthers] = others
+	}
+	if len(others.TimestampList) == 0 {
+		others.TimestampList = []uint64{timestamp}
+		others.CPUTimeMsList = []uint32{totalCPUTimeMs}
+	} else {
+		others.TimestampList = append(others.TimestampList, timestamp)
+		others.CPUTimeMsList = append(others.CPUTimeMsList, totalCPUTimeMs)
+	}
+	others.CPUTimeMsTotal += uint64(totalCPUTimeMs)
+}
+
+func (d *dataPoints) merge(other *dataPoints) {
+	if other == nil || len(other.TimestampList) == 0 {
+		return
+	}
+	if other.isInvalid() {
+		logutil.BgLogger().Warn("[top-sql] data points is invalid, it should never happen", zap.Any("self", d), zap.Any("other", other))
+		return
+	}
+	// Sort the dataPoints by timestamp to fix the affect of time jump backward.
+	sort.Sort(d)
+	sort.Sort(other)
+	if len(d.TimestampList) == 0 {
+		d.TimestampList = other.TimestampList
+		d.CPUTimeMsList = other.CPUTimeMsList
+		d.CPUTimeMsTotal = other.CPUTimeMsTotal
+		return
+	}
+	length := len(d.TimestampList) + len(other.TimestampList)
+	timestampList := make([]uint64, 0, length)
+	cpuTimeMsList := make([]uint32, 0, length)
+	i := 0
+	j := 0
+	for i < len(d.TimestampList) && j < len(other.TimestampList) {
+		if d.TimestampList[i] == other.TimestampList[j] {
+			timestampList = append(timestampList, d.TimestampList[i])
+			cpuTimeMsList = append(cpuTimeMsList, d.CPUTimeMsList[i]+other.CPUTimeMsList[j])
+			i++
+			j++
+		} else if d.TimestampList[i] < other.TimestampList[j] {
+			timestampList = append(timestampList, d.TimestampList[i])
+			cpuTimeMsList = append(cpuTimeMsList, d.CPUTimeMsList[i])
+			i++
+		} else {
+			timestampList = append(timestampList, other.TimestampList[j])
+			cpuTimeMsList = append(cpuTimeMsList, other.CPUTimeMsList[j])
+			j++
+		}
+	}
+	if i < len(d.TimestampList) {
+		timestampList = append(timestampList, d.TimestampList[i:]...)
+		cpuTimeMsList = append(cpuTimeMsList, d.CPUTimeMsList[i:]...)
+	}
+	if j < len(other.TimestampList) {
+		timestampList = append(timestampList, other.TimestampList[j:]...)
+		cpuTimeMsList = append(cpuTimeMsList, other.CPUTimeMsList[j:]...)
+	}
+	d.TimestampList = timestampList
+	d.CPUTimeMsList = cpuTimeMsList
+	d.CPUTimeMsTotal += other.CPUTimeMsTotal
+}
+
+func addEvictedDataPoints(others *dataPoints, evict *dataPoints) *dataPoints {
+	if others == nil {
+		others = &dataPoints{}
+	}
+	others.merge(evict)
+	return others
```

Comment:
This function is called only once, maybe not necessary to be a function? You will find the code **more readable** if you remove this function.